src/components/Card.js

```diff
@@ -16,31 +25,53 @@ export default class Card {
     return cardElement;
   }
 
-  _handleDeleteCard() {
+  getId() {
+    return this._id;
+  }
+
+
+  removeCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
   _handleLikeIcon() {
-    this._cardElement.querySelector(".card__like-button").classList.toggle("card__like-button_active");
+    this._likeButton.classList.toggle("card__like-button_active");
+    if (this._cardElement.querySelector(".card__like-button_active")) {
+      this._api.likeCard(this.getId());
+    } else {
+      this._api.unlikeCard(this.getId());
+    }
+  }
+
+  _setLike() {
+    if (this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    }
   }
 
+
   _setEventListeners() {
     this._likeButton.addEventListener("mousedown", () => this._handleLikeIcon());
-    this._deleteButton.addEventListener("mousedown", () => this._handleDeleteCard());
+    this._binIcon.addEventListener("mousedown", () => {
+      this._handleDelete(this);
+    });
+
     this._cardImage.addEventListener("mousedown", () => {
       this._handleImageClick({ link: this._link, name: this._name });
     });
   }
 
   generateCard() {
     this._cardElement = this._getTemplate();
+
     this._likeButton = this._cardElement.querySelector(".card__like-button");
```

Comment:
Good job that you've found these variables only once here and defined them as **class fields**